maxstertrooper/js/logic.js

```diff
@@ -4,16 +4,35 @@ class Runner {
   constructor(canvas, context) {
     this.canvas = canvas;
     this.context = context;
+    this.height = 150;
+    this.width = 300;
     this.flyers = {};
     this.bullets = {};
-    this.cannon = new Cannon(canvas.width, canvas.height);
+    this.cannon = new Cannon(this.width, this.height);
     this.display = new PointsDisplay();
     this.points = 15;
     this.loopingInverval = null;
     this.gameIsOn = false;
     this.menuDisplay = new MenuDisplay();
   }
 
+  scale(smooth=true) {
```

Comment:
the Runner class responsability was to run the game (start, pause, manage menu display, and handle the game loop)

adding Display management to it (scale function) gives it a second responsability: which is not desired

what would you think of managing the Display outside of the Game Runner logic?

Reply:
OK